proxy/src/lib.rs

```diff
@@ -343,20 +345,24 @@ fn serve<R, B, E, F, G>(
     sensors: telemetry::Sensors,
     get_orig_dst: G,
     drain_rx: drain::Watch,
-    executor: &Handle,
-) -> Box<Future<Item = (), Error = io::Error> + 'static>
+) -> Box<Future<Item = (), Error = io::Error> + Send + 'static>
 where
-    B: tower_h2::Body + Default + 'static,
-    E: Error + 'static,
-    F: Error + 'static,
+    B: tower_h2::Body + Default + Send + 'static,
+    <B::Data as ::bytes::IntoBuf>::Buf: Send,
+    E: Error + Send + 'static,
+    F: Error + Send + 'static,
     R: Recognize<
         Request = http::Request<HttpBody>,
         Response = http::Response<telemetry::sensor::http::ResponseBody<B>>,
         Error = E,
         RouteError = F,
     >
-        + 'static,
-    G: GetOriginalDst + 'static,
+        + Send + Sync + 'static,
+    R::Key: Send,
+    R::Service: Send,
```

Comment:
it seems likely that these bounds should just be added into the `Recognize` trait; but probably for a followup.